### PR TITLE
[ADD] Added member APIs

### PR DESCRIPTION
**Pull request by rakibul-islam-raju**

### CODE REVIEW

**Comment by rakibul-islam-raju** [comment]
#15

**Comment by rakibul-islam-raju** [comment]
#16

File: backend/peoples/serializers.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    class Meta:
+        model = Member
+        fields = [
+            "name",
+            "mobile_number",
+            "nid_number",
+            "guardian_name",
+            "gender",
+            "serial_number",
+            "uuid",
```
Comment (harunurkst) [inline backend/peoples/serializers.py:16]: uuid should be auto-generated,

File: backend/peoples/serializers.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            "team",
+            "branch",
```
Comment (harunurkst) [inline backend/peoples/serializers.py:18]: branch value should be logged in staff's branch. No need to provide value from frontend.

File: backend/peoples/serializers.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        model = Member
+        fields = [
+            "name",
+            "mobile_number",
+            "nid_number",
+            "guardian_name",
+            "gender",
+            "serial_number",
+            "team",
+            "branch",
```
Comment (harunurkst) [inline backend/peoples/serializers.py:34]: branch value should be logged-in staff's branch.

File: backend/peoples/views.py [hunk 4]
```diff
@@ -1,2 +1,28 @@
...
+
+    def get_serializer_class(self):
+        if self.request.method == "POST":
+            return MemberCreateSerializer
+        return MemberDetailSerializer
+
+
+class MemberDetailsView(RetrieveUpdateDestroyAPIView):
+    permission_classes = [IsAuthenticated]
+    queryset = Member.objects.all()
```
Comment (harunurkst) [inline backend/peoples/views.py:27]: show only logged-in staff's branch member

**Review by harunurkst — CHANGES_REQUESTED** [review]
please resolve my feedback

**Comment by rakibul-islam-raju** [comment]
Thanks for your feedback! I've made the requested changes and pushed the updates to this PR. Please take another look and let me know if there's anything else that needs to be addressed.

**Pull request merged**